src/main/kotlin/com/carneirofatima/vehicleterminal/controller/DriverController.kt

```diff
@@ -0,0 +1,46 @@
+package com.carneirofatima.vehicleterminal.controller
+
+import com.carneirofatima.vehicleterminal.ValidationRequest
+import com.carneirofatima.vehicleterminal.service.DriverService
+import com.carneirofatima.vehicleterminal.to.DriverTO
+import org.springframework.http.HttpEntity
+import org.springframework.http.ResponseEntity
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.PutMapping
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RestController
+import javax.validation.Valid
+
+@RestController
+@RequestMapping("/driver")
+class DriverController(
+    private val service: DriverService,
+    private val validator: ValidationRequest
+) {
+
+    @PostMapping
+    fun save(
+       @Valid @RequestBody form: DriverTO
+    ): HttpEntity<DriverTO> {
+        validator.validateSave(form)
```

Comment:
if you need to add from queue, you will need to revalidate on consumer. Perhaps is better put validations inside service method